# 01 · Business Understanding

**Phase goal:** frame the problem in *business* terms and define the measurable KPIs a model must clear before it can ship.

## The problem
A used-car marketplace needs an **instant, trustworthy price estimate** for any car a seller lists, plus a **budget band** (Low / Medium / High) for search and filtering. Manual appraisal is slow and inconsistent; a model that is *fast, accurate and explainable* directly improves listing conversion and pricing confidence.

## Stakeholders & value
| Stakeholder | What they need | KPI it maps to |
| :-- | :-- | :-- |
| Seller | A fair, believable price | Low typical error (MAE in ₹) |
| Buyer | Sensible budget bands | Band accuracy |
| Product | Confidence to auto-suggest | R² / calibration |
| Engineering | Something cheap to serve | Small, fast artifact |

## The ML task
A **regression** predicting the exact price (in ₹ Lakhs). The **Low/Medium/High band is *derived* from the predicted price**, so the band and the number can never disagree — a deliberate simplification over training a separate, less-accurate classifier.

See [`docs/BUSINESS_CASE.md`](../docs/BUSINESS_CASE.md) for the full write-up.

In [1]:
import sys, warnings
sys.path.insert(0, '../src')
warnings.filterwarnings('ignore')
from car_pricing import config
print('KPI gates a model must clear to ship:')
for k, v in config.KPI.items():
    print(f'  {k:20s} {v}')
print('\nPrice unit:', config.PRICE_UNIT, '| Bands:', config.BAND_LABELS)

KPI gates a model must clear to ship:
  max_mae_lakhs        1.0
  min_r2               0.85
  min_band_accuracy    0.7

Price unit: Lakhs | Bands: ['Low', 'Medium', 'High']


## Success criteria (the ship/no-ship gate)

The pipeline **automatically refuses to ship** a model that fails any of:
- **MAE ≤ ₹1.0 Lakh** — the typical miss must be under ₹1,00,000.
- **R² ≥ 0.85** — explain at least 85 % of price variance.
- **Band accuracy ≥ 70 %** — correct budget bucket most of the time.

## SDLC roadmap (the rest of these notebooks)
1. Business understanding *(this notebook)*
2. Data understanding & EDA
3. Data-format benchmarks (CSV vs Parquet vs Feather)
4. Feature engineering (target encoding)
5. Model comparison (DT vs RF vs XGBoost vs LightGBM …)
6. Evaluation & selection
7. Productionisation (the shippable pipeline)